tempel.el

```diff
@@ -524,7 +560,13 @@ This is meant to be a source in `tempel-template-sources'."
   (cl-loop for i below (abs arg) do
            (if-let (next (tempel--find arg)) (goto-char next)
              (tempel-done)
-             (cl-return))))
+             (cl-return)))
+
+  ;; If the current field or location marker is marked as "terminating",
+  ;; disable the currently active template right away.
+  (when-let (ov (or (tempel--field-at-point) (tempel--location-marker-at-point)))
+    (when (tempel--prop-get 'terminate ov)
+      (tempel--disable))))
```

Comment:
I am not sure about the logic here. I think it would make more sense to change `tempel--find` to return the actual tempel field which has been found and then terminate the corresponding template instead of terminating the topmost template. tempel--disable should probably take an optional template argument and take the topmost template otherwise.